### PR TITLE
add a .first method to RecordCollections

### PR DESCRIPTION
**Pull request by whit537**
It's a common use-case to want one and only one result from a query.

### CODE REVIEW

**Comment by whit537** [comment]
Includes #42.

**Comment by whit537** [comment]
Rebased to fix a broken test in the appropriate commit, previous head was dcb9e7a9be11265f1ec15f633051bfb63f139fc8.

**Comment by whit537** [comment]
499fd5bdaa16485cdb86ee98f082210b6d82b370 allows for passing in an exception as `default` and having that be raised for you. Not sure if you'll want that feature even if you want the rest of the PR.

**Comment by whit537** [comment]
Ready for review, @kennethreitz. :)

**Comment by kennethreitz** [comment]
So, I have mixed feelings about this one.

Currently, you can get this behavior with `rows[0]` (get just the first row), or `rows[0:1].all(as_dict=True)`.
I do see some API symmetry between `one` & `all`, however. 

I tend to lean on the side of introducing as few methods as possible to an API, and adding only what is necessary. This does seem like a good nice-to-have, though.

I am torn.

File: records.py [hunk 1]
```diff
@@ -189,6 +201,33 @@ def all(self, as_dict=False, as_ordereddict=False):
 
         return rows
 
+    def one(self, default=None, as_dict=False, as_ordereddict=False):
+        """Returns a single record for the RecordCollection, or `default`."""
+
+        # Try to get a record, or return default.
```
Comment (kennethreitz) [inline records.py]: This could be simplified to simply use use 
```python
record = self[0]
```

Comment (whit537) [inline records.py]: Done in 5e60bb96cc8970f1ca036d332a341f3b89514546.

**Comment by whit537** [comment]
Rebased on master at dd440860c86762480a7bd14f7f0ffd74408cefc3. Previous head was 5e60bb96cc8970f1ca036d332a341f3b89514546.

**Comment by whit537** [comment]
> Currently, you can get this behavior with `rows[0]` (get just the first row), or `rows[0:1].all(as_dict=True)`.

Yes, but quite often it should be a bug if there's more than one result, and/or not a bug if there's less than one. The purpose of the `one` method is to provide this error checking and defaulting.

> I am torn.

Well, it's here if you want it. :-)

**Comment by Brobin** [comment]
Would `first` be a better name? It's a little more specific to which one you are retrieving, and similar to other DB/ORM Api's I've seen before.

I do agree that just doing `rows[0]` is already pretty easy, and it's not that difficult.

**Comment by whit537** [comment]
`first` makes it sound like you're expecting more than one, and you just want the first one, in which case there's not much advantage over `rows[0]`.

`one` suggests that for the query to return more than one result is a bug.

**Comment by Brobin** [comment]
Ah, gotcha. I didn't realize you were checking if there was more than one record.

**Comment by whit537** [comment]
And also [defaulting](https://github.com/kennethreitz/records/blob/a49d1ff0e9507c8047a9693fc7f6a09218c026ed/records.py#L207-L213) if there's less than one. :)

**Comment by kennethreitz** [comment]
@Brobin that was what I thought to. That's one of the main reasons I'm not gung ho about adding this — I think it's confusing :)

`RecordSet.exactly_one()` on the other hand, hahaha.

**Comment by Brobin** [comment]
`RecordSet.exactly_one_and_fail_if_there_are_more()` :stuck_out_tongue_winking_eye:

**Comment by kennethreitz** [comment]
I think `.first()` is what I want, and would be nice to have. It would return `None` if none exists, I think.

this needs a rebase if we wish to proceed ;)

**Comment by whit537** [comment]
Rebased on `master` at 9e6583311756a94ca20572b96063c6e920b388e0. Previous head was a49d1ff0e9507c8047a9693fc7f6a09218c026ed. Renaming to first ... :-)

**Comment by whit537** [comment]
No CI? I will run tests locally ...

**Comment by whit537** [comment]
85fbec787a71f4f97c61211ad2ec256cb5c8bcdc

```
(env) $ py.test tests/
============================================ test session starts =============================================
platform darwin -- Python 3.5.0, pytest-2.8.7, py-1.4.31, pluggy-0.3.1
rootdir: /Users/whit537, inifile: pytest.ini
collected 15 items 

tests/test_records.py .............
tests/test_transactions.py ..

========================================= 15 passed in 1.47 seconds ==========================================
(env) $
```

**Comment by whit537** [comment]
3b7438f130fe2b37709a5ce9b08d4fa73e007e50

```
(env) $ py.test tests/
============================================ test session starts =============================================
platform darwin -- Python 3.5.0, pytest-2.8.7, py-1.4.31, pluggy-0.3.1
rootdir: /Users/whit537, inifile: pytest.ini
collected 15 items 

tests/test_records.py .............
tests/test_transactions.py ..

========================================= 15 passed in 1.42 seconds ==========================================
(env) $
```

**Comment by whit537** [comment]
Alright, @kennethreitz, lemme know if you want a squash or further changes or whatevs. :-)

**Comment by kennethreitz** [comment]
✨🍰✨

**Pull request merged**